### DISCUSSION TITLE
关于3D工业相机的Z轴精度的疑问

### DISCUSSION THREAD

**Discussion by JOJO-K-DQ** [Open3DV/Xema#21] [Q&A] [answered]
各位工程师们好！很高兴能看到有一家企业能愿意整合制造业优势资源，构建工业视觉领域开源社区，推动社区成为可持续发展的自组织系统，为社会做出自己的贡献，我是一个刚刚入职这个行业小白，有一些问题想请教前辈们，如果前辈们在空闲时间能给我这个小辈解答一下真的感激不尽。

我想问的问题：我们在制作一款工业相机的之前或者在写技术方案的时候肯定会对相机的Z轴的精度做一个理论上的计算，然后在制作完相机之后需要对相机的Z轴的精度进行测算，所以我想问的是，是否有相关的理论(计算公式、数学模型)或者证明材料能说明相机在Z轴上的精度为多少？

**Comment by vic-w** [comment, marked as answer]
实际上，有一个标准vdi/vde 2634 part2，里面有描述测量精度的方法。我们的XEMA相机暂时还没有以这个标准测量的数据。这个标准里面定义了三种误差表示方式：
1. 球直径探测误差
2. 球心距测量误差
3. 平面度测量误差
这些都不能简单等同于人们说的”Z轴精度“，那到底什么是”Z轴精度“似乎仍然没有人给出定义。
我们的相机标注的Z轴的精度实际上是一个测试的数值。我们测试的工具是标定板。相机拍摄标定板上标志点之间的距离与真实的距离的偏差，换算成Z轴的偏差，就得到我们标注的精度。这个精度值基本上是给工程人员看的，在相机选型时有一定的指导意义。

如果要从理论上讨论，我觉得首先要区分我们要讨论的是**系统误差**还是**随机误差**。

系统误差是不管测量多少次都会产生的误差，这是理论模型和实际物理参数之间的偏差引起的。可以通过改进理论模型来进行消除系统误差。
产生系统误差原因包括：
1. 镜头的针孔+畸变的模型不能完全描述镜头的物理参数
2. 相机镜头有模糊的现象，像素的亮度受到了其周围像素的影响
3. 圆点标定板在拍摄时会产生的椭圆圆心偏差
4. 镜头在标定后，由于震动产生了位移
5. DLP投影的条纹并非真正地以正弦的形式变化

随机误差是每次拍摄都会有不同的。可以通过多次拍摄取平均来消除随机误差。
产生随机误差的原因包括：
1. 相机像素的热噪声
2. 投影仪像素的随机噪声（可能也算是热噪声？）
3. 环境光线变化的影响

如果你想要推算理论上能达到多少精度，那就首先要用下图这个双目视觉的模型，把像素坐标和z坐标之间的关系
![image](https://github.com/Open3DV/Xema/assets/3446495/afff0b77-8468-4f19-88e5-fbde89dae0ef)
然后再把像素坐标和像素的亮度关联起来。这样就可以推算出由亮度变化所导致的误差程度。

而关于相机镜头标定和标定板圆心造成的误差，我们尝试用渲染的方式来模拟真实情况来估计误差。
我们实验的代码在这个工程：https://github.com/Open3DV/calibration_by_rendering
这个程序在空间中模拟了一个双目相机，悬挂在距地面一定距离的高处。并渲染出标定板在各种位置和姿态下，分别在双目相机中的成像。使用这个程序可以估计出圆点标定板椭圆圆心偏差造成的影响。如果你有需要可以尝试运行一下。

总之，3D相机的误差是有很多因素组成的，我目前还没有看到有全面的分析。上面也只是我的一个思路和一些初步的尝试，我们可以再进一步的讨论。
